src/components/LayoutTree.tsx

```diff
@@ -414,6 +417,8 @@ const TreeNode = ({
 } & Pick<CSSTransitionProps, 'in'>) => {
   const [y, setY] = useState(_y)
   const fadeThoughtRef = useRef<HTMLDivElement>(null)
+  const lastPatches = useSelector(state => state.undoPatches[state.undoPatches.length - 1])
+  const isLastActionNewThought = lastPatches?.filter(patch => patch.actions[0] === 'newThought')
```

Comment:
Since `.filter` leaves `isLastActionNewThought` as an array, it will always be truthy even if empty (i.e. where we use it in the ternary below to determine `layoutTransition`).

```suggestion
  const isLastActionNewThought = lastPatches?.some(patch => patch.actions[0] === 'newThought')
```

Reply:
I've updated the logic with `.some` but didn't experience that lagging problem.
[screencast-localhost_3000-2024_12_12-07_25_33.webm](https://github.com/user-attachments/assets/103cac82-6fbd-442a-8049-fbffff0580b1)

I'll double check. 

I'm wondering, is it happening on the main branch too? @trevinhofmann

Reply:
On `main`, we do have the overlap due to the animation being too slow (which this PR addresses):

https://github.com/user-attachments/assets/506cc0c6-f3d0-4af2-a90d-9c405a7bdd72

However, I don't experience the same lag issue which is still happening on this PR's branch:

https://github.com/user-attachments/assets/63aaaaff-a567-4eb4-8e50-f9e0202ed0cb

It isn't super consistent, but happens fairly often for me. Try holding the `enter` key to create new thoughts more quickly than just clicking the toolbar button.

Reply:
I can confirm that the issue mentioned above can also be reproduced in the main branch. 
I doubt this is related to the changes in this PR, but I'll figure out a solution to fix it.

https://github.com/user-attachments/assets/a34c887a-76c2-479b-b9a8-6e4663afd49d

Reply:
That is interesting. I'll need to retest, because I hadn't seen it in `main` before. If it is happening in `main` and not a new regression, I'd rather not fix it in this PR.

Reply:
I'm still unable to reproduce on `main`:

https://github.com/user-attachments/assets/fe7e2fc0-afaa-4b70-a53e-b843c35abd3b

And seeing it consistently on `fix/fade-in-new-thought`:

https://github.com/user-attachments/assets/9cf7c842-f9f2-4cef-83b2-b1188a9e34aa